charactersheet/charactersheet/viewmodels/dm/encounter_sections/point_of_interest_section.js

```diff
@@ -0,0 +1,171 @@
+        var poi = PersistenceService.findBy(PointOfInterest, 'encounterId', self.encounterId());
+        if (poi) {
+            self.pointsOfInterest(poi);
+        }
+
+        var section = PersistenceService.findFirstBy(PointOfInterestSection, 'encounterId', self.encounterId());
+        if (!section) {
+            section = new PointOfInterestSection();
+            section.encounterId(self.encounterId());
+            section.characterId(key);
+        }
+        self.name(section.name());
+        self.visible(section.visible());
+    };
+
+    self.unload = function() {
+
+    };
+
+    self.save = function() {
+        var key = CharacterManager.activeCharacter().key();
+        var section = PersistenceService.findFirstBy(PointOfInterestSection, 'encounterId', self.encounterId());
+        if (!section) {
+            section = new PointOfInterestSection();
+            section.encounterId(self.encounterId());
+            section.characterId(key);
+        }
+
+        section.name(self.name());
+        section.visible(self.visible());
+        section.save();
+
+        self.pointsOfInterest().forEach(function(poi, idx, _) {
+            poi.save();
+        });
+    };
+
+    self.delete = function() {
+        var section = PersistenceService.findFirstBy(PointOfInterestSection, 'encounterId', self.encounterId());
+        if (section) {
+            section.delete();
+        }
+
+        self.pointsOfInterest().forEach(function(poi, idx, _) {
+            poi.delete();
+        });
+    };
+
+    /* UI Methods */
+
+    /**
+     * Filters and sorts the weaponss for presentation in a table.
+     */
+    self.filteredAndSortedPointsOfInterest = ko.computed(function() {
+        return SortService.sortAndFilter(self.pointsOfInterest(), self.sort(), null);
+    });
+
+    /**
+     * Determines whether a column should have an up/down/no arrow for sorting.
+     */
+    self.sortArrow = function(columnName) {
+        return SortService.sortArrow(columnName, self.sort());
+    };
+
+    /**
+     * Given a column name, determine the current sort type & order.
+     */
+    self.sortBy = function(columnName) {
+        self.sort(SortService.sortForName(self.sort(), columnName, self.sorts));
+    };
+
+    self.addPointOfInterest = function() {
+        var poi = self.blankPointOfInterest();
+        poi.characterId(CharacterManager.activeCharacter().key());
+        poi.encounterId(self.encounterId());
+        poi.save();
+        self.pointsOfInterest.push(poi);
+        self.blankPointOfInterest(new PointOfInterest());
+    };
+
+    self.removePointOfInterest = function(poi) {
+        poi.delete();
+        self.pointsOfInterest.remove(poi);
+    };
+
+    self.editPointOfInterest = function(poi) {
+        self.selecteditem(poi);
+    };
+
+    self.toggleModal = function() {
+        self.openModal(!self.openModal());
+    };
+
+    /* Modal Methods */
+
+    self.modalFinishedOpening = function() {
```

Comment:
I guess I added them because I thought I needed them, but then didn't use them. I'm not sure if this is good/bad.